### PR TITLE
feat(match collection): add locking endpoint

### PR DESCRIPTION
**Pull request by AdrianAndersen**

### CODE REVIEW

**Comment by AdrianAndersen** [comment]
[Mulighet for å låse/åpne overleveringer for brukere](https://trello.com/c/b8jlQD4k/1058-mulighet-for-%C3%A5-l%C3%A5se-%C3%A5pne-overleveringer-for-brukere)

File: src/collections/match/operations/match-lock.operation.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+export interface MatchLockSpec {
+  customer: string;
+  lock: boolean;
+}
+
+export function verifyMatchLockSpec(
+  matchLockSpec: unknown,
+): matchLockSpec is MatchLockSpec {
+  const m = matchLockSpec as Record<string, unknown>;
```
Comment (LarsSelbekk) [inline src/collections/match/operations/match-lock.operation.ts:25]:
```suggestion
  const m = matchLockSpec as Record<string, unknown> | null | undefined;
```
and maybe extract a type for this as we should be using this type more generally

File: src/collections/match/operations/match-lock.operation.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  return (
+    m &&
```
Comment (LarsSelbekk) [inline src/collections/match/operations/match-lock.operation.ts:27]:
```suggestion
    !!m &&
```

File: src/collections/match/operations/match-lock.operation.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    isBoolean(m["lock"])
+  );
+}
+
+export class MatchLockOperation implements Operation {
+  constructor(private matchStorage?: BlDocumentStorage<Match>) {
+    this.matchStorage = matchStorage
+      ? matchStorage
+      : new BlDocumentStorage(BlCollectionName.Matches, matchSchema);
+  }
```
Comment (LarsSelbekk) [inline src/collections/match/operations/match-lock.operation.ts:39]: Should use separate declaration of the field so TypeScript knows it's not null - see order-place.operation.ts

**Review by LarsSelbekk — CHANGES_REQUESTED** [review]

**Review by LarsSelbekk — APPROVED** [review]

**Pull request merged**